check_cache.py

```diff
@@ -0,0 +1,227 @@
+"""
+Don't run this script directly, let the CI invoke it and determine whether a
+workflow should run or not.
```

Comment:
The plan is to have a workflow that can only be manually dispatched called "Purge CI Cache" and it will properly invoke `check_cache.py` with the `--purge-cache` flag and provide the necessary Github Tokens with permissions to delete repo vars.

You can, theoretically, still run this script manually but I think the comment should remain. It works best when a Github action is the user.

Reply:
I've made the comment clearer. resolving.